### PR TITLE
BigQuery: rewrite simple app tutorial.

### PR DESCRIPTION
**Pull request by tswast**
Rewrite of the sample used at
https://cloud.google.com/bigquery/create-simple-app-api

- Add additional region tags to support tutorial rewrite.
- Use more relevant query from public datasets.
- Remove extra classes, making the sample more similar to the
  quickstart.
- Add a test for this sample.

### CODE REVIEW

**Comment by tswast** [comment]
Note: I've already frozen the docs at https://cloud.google.com/bigquery/create-simple-app-api to a commit hash, so this should be safe to merge whenever.

File: bigquery/stackoverflow.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+
+# Copyright 2015 Google, Inc.
```
Comment (frankyn) [inline bigquery/stackoverflow.rb]: Update year.

File: bigquery/stackoverflow.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
```
Comment (frankyn) [inline bigquery/stackoverflow.rb]: Remove this line.

File: bigquery/stackoverflow.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+# [END bigquery_simple_app_deps]
+
+# [START bigquery_simple_app_client]
+# This uses Application Default Credentials to authenticate.
+# @see https://cloud.google.com/bigquery/docs/authentication/getting-started
+bigquery = Google::Cloud::Bigquery.new
+# [END bigquery_simple_app_client]
+
+# [START bigquery_simple_app_query]
+sql = <<-END_SQL
```
Comment (frankyn) [inline bigquery/stackoverflow.rb]: Update sql variable to use concat strings instead of using heredoc.

File: bigquery/spec/stackoverflow_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Copyright 2016 Google, Inc
```
Comment (frankyn) [inline bigquery/spec/stackoverflow_spec.rb]: Update year.

File: bigquery/stackoverflow.rb (hunk 3 above)
Comment (tswast) [reply bigquery/stackoverflow.rb]: Done.

File: bigquery/stackoverflow.rb (hunk 1 above)
Comment (tswast) [reply bigquery/stackoverflow.rb]: Updated to 2017. Also changed to Google LCC per go/releasing/preparing

File: bigquery/stackoverflow.rb [hunk 5]
```diff
@@ -28,17 +26,13 @@
...
-  view_count
-FROM `bigquery-public-data.stackoverflow.posts_questions`
-WHERE tags like '%google-bigquery%'
-ORDER BY view_count DESC
-LIMIT 10
-END_SQL
+sql = "SELECT " +
+      "CONCAT('https://stackoverflow.com/questions/', " +
+      "       CAST(id as STRING)) as url, " +
+      "view_count " +
```
Comment (frankyn) [inline bigquery/stackoverflow.rb]: Question: `view_count` should be able to fit on the previous line. Is there a reason to keep it on a separate line?

Comment (tswast) [reply bigquery/stackoverflow.rb]: not really.

File: bigquery/stackoverflow.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+# [END bigquery_simple_app_deps]
+
+# [START bigquery_simple_app_client]
+# This uses Application Default Credentials to authenticate.
+# @see https://cloud.google.com/bigquery/docs/authentication/getting-started
+bigquery = Google::Cloud::Bigquery.new
+# [END bigquery_simple_app_client]
+
+# [START bigquery_simple_app_query]
+sql = "SELECT " +
```
Comment (frankyn) [inline bigquery/stackoverflow.rb]: One last nit, could you align the `=` for sql and results assignments.

Comment (tswast) [reply bigquery/stackoverflow.rb]: sure, done.

**Comment by frankyn** [comment]
Please wait until tests complete to merge. Sample LGTM! Thanks for your patience!

**Pull request merged**